### ISSUE TITLE
Issue when rendering SDF file with jinja conditionals

### ISSUE THREAD

**Issue by ryankwilliams** [RedHatQE/teflo#132]
I found a bug where teflo validation can't run when I have jinja conditionals mixed within my include statements. These conditionals are here to only include those SDF files when the required conditional are met.

I put together this simple Python script that pretty much takes the teflo function for jinja template and allows me to quickly run it to demonstrate this issue.

**Script**

```python
import jinja2
import os
import yaml

def template_render(filepath, env_dict):
    """
    A function to do jinja templating given a file and a dictionary of key/vars
    :param filepath: path to a file
    :param env_dict: dictionary of key/values used for data substitution
    :return: stream of data with the templating complete
    :rtype: data stream
    """
    path, filename = os.path.split(filepath)
    return jinja2.Environment(loader=jinja2.FileSystemLoader(
        path), lstrip_blocks=True, trim_blocks=True).get_template(filename).render(env_dict)

data = template_render("libvirt.yml", {"SOME_ACTION": True})
print(data)
print("\n")
data_yaml = yaml.safe_load(data)
print(data_yaml)
```

**Issue**

I have the following SDF file:

```yaml
---
name: libvirt_stack
description: >
  Teflo scenario descriptor file declaring the components required
  to deploy a development stack for development purposes. The
  development stack is a bare metal machine, with libvirt vms deployed
  on top and then OpenShift, CNV, ODF installed and configured.

include:
  - teflo/stack/provision_localhost.yml
  - teflo/stack/provision_libvirt.yml
{% if SOME_ACTION is defined and SOME_ACTION %}  - teflo/stack/orchestrate-123.yml{% endif %}
  - teflo/stack/orchestrate.yml
```

When jinja template is performed and SOME_ACTION is True, the result is the following:

```yaml
---
name: libvirt_stack
description: >
  Teflo scenario descriptor file declaring the components required
  to deploy a development stack for development purposes. The
  development stack is a bare metal machine, with libvirt vms deployed
  on top and then OpenShift, CNV, ODF installed and configured.

include:
  - teflo/stack/provision_localhost.yml
  - teflo/stack/provision_libvirt.yml
  - teflo/stack/orchestrate-123.yml  - teflo/stack/orchestrate.yml


{'name': 'libvirt_stack', 'description': 'Teflo scenario descriptor file declaring the components required to deploy a development stack for development purposes. The development stack is a bare metal machine, with libvirt vms deployed on top and then OpenShift, CNV, ODF installed and configured.\n', 'include': ['teflo/stack/provision_localhost.yml', 'teflo/stack/provision_libvirt.yml', 'teflo/stack/orchestrate-123.yml  - teflo/stack/orchestrate.yml']}
```
You can see that both SDF files are included on the same line (same string) and when run from teflo it fails right away not even allowing me to validate it.

```
2021-09-24 17:50:55,576 WARNING Scenario workspace was not set, therefore the workspace is automatically assigned to the current working directory. You may experience problems if files needed by teflo do not exists in the scenario workspace.
Included File is invalid or Include section is empty.You have to provide valid scenario files to be included.
```

The workaround for now is to add a new line between included SDF files in order for it to work with the jinja rendering.

**Comment by dno-github** [comment]
Jira ticket CCITCARBON-370 has been created for this issue. Url : https://issues.redhat.com/browse/CCITCARBON-370

**Comment by rujutashinde** [comment]
@ryankwilliams You are seeing this issue with teflo 2.0 or prior releases?
Was there any release that the jinja templating with include worked?

**Comment by ryankwilliams** [comment]
Hi @rujutashinde, I was seeing this issue with 2.0 and before. I could use one jinja conditional block for an `include` statement and it works. Its when you have multiple conditionals, I run into the issue above.

**Comment by JunqiZhang0** [comment]
closed by pr #153
